src/vcpkg/dependencies.cpp

```diff
@@ -837,16 +841,58 @@ namespace vcpkg::Dependencies
                                                     spec.name(),
                                                     spec.feature());
                         paragraph_depends = &maybe_paragraph.value_or_exit(VCPKG_LINE_INFO).dependencies;
+                        has_supports = !maybe_paragraph.get()->supports_expression.is_empty();
                     }
 
                     // And it has at least one qualified dependency
-                    if (paragraph_depends &&
-                        Util::any_of(*paragraph_depends, [](auto&& dep) { return !dep.platform.is_empty(); }))
+                    if (has_supports || (paragraph_depends && Util::any_of(*paragraph_depends, [](auto&& dep) {
+                                             return !dep.platform.is_empty();
+                                         })))
                     {
                         // Add it to the next batch run
                         qualified_dependencies.emplace_back(spec);
                     }
                 }
+                else
+                {
+                    const PlatformExpression::Expr* supports_expression = nullptr;
+                    if (spec.feature() == "core")
+                    {
+                        supports_expression =
+                            &clust.get_scfl_or_exit().source_control_file->core_paragraph->supports_expression;
+                    }
+                    else if (spec.feature() != "default")
+                    {
+                        auto maybe_paragraph =
+                            clust.get_scfl_or_exit().source_control_file->find_feature(spec.feature());
+                        Checks::check_maybe_upgrade(VCPKG_LINE_INFO,
+                                                    maybe_paragraph.has_value(),
+                                                    "Package %s does not have a %s feature",
+                                                    spec.name(),
+                                                    spec.feature());
+
+                        supports_expression = &maybe_paragraph.get()->supports_expression;
+                    }
```

Comment:
It looks like this could be extracted into a function like this?:

```
const PlatformExpression::Expr* get_applicable_supports_expression(Cluster& clust, FeatureSpec spec)
```

?

Reply:
I made this a member function of the cluster